packages/edit-site/src/components/sidebar-navigation-screen-pages/index.js

```diff
@@ -35,65 +41,85 @@ export default function SidebarNavigationScreenPages() {
 		'page'
 	);
 
-	const { setIsCreatePageModalOpened } = useDispatch( editSiteStore );
+	const [ showAddPage, setShowAddPage ] = useState( false );
+
+	const history = useHistory();
+	const { setPage } = unlock( useDispatch( editSiteStore ) );
+
+	const handleNewPage = ( { type, id } ) => {
+		setPage( {
+			context: { postType: type, postId: id },
+		} );
+		// Navigate to the created template editor.
+		history.push( {
```

Comment:
Can we keep this callback within the modal component maybe?

Reply:
Went back and forth on this and ended up in this approach as it felt a little more flexible but happy to change if you'd prefer to move it in modal.

Reply:
I'm not strongly opinionated either. I can see arguments for both.